tqdm/_tqdm.py

```diff
@@ -133,10 +146,23 @@ def tqdm(iterable, desc=None, total=None, leave=False, file=sys.stderr,
     else:
         dynamic_miniters = False
 
+    if ascii is None:
+        try:
+            file.encoding
+        except AttributeError:
+            ascii = True
+        else:
+            if file.encoding:
+                ascii = not ('U8' == file.encoding or
+                             ('utf' in file.encoding) or
+                             ('UTF' in file.encoding))
+            else:
+                ascii = True
```

Comment:
will leave it to you ;) i'm not pedantic about loop startup overhead so don't mind abstractions at this level.